packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -422,18 +478,78 @@ export async function askUserConfirm(
     return false;
   }
   sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotification, {
-    [TelemetryProperty.Status]: ProjectMigratorStatus.OK,
+    [TelemetryPropertyKey.status]: TelemetryPropertyValue.ok,
+    [TelemetryPropertyKey.mode]: TelemetryPropertyValue.modal,
   });
   return true;
 }
 
-export async function popupMessage(
+export async function showNonmodalNotification(
+  ctx: CoreHookContext,
+  versionForMigration: VersionForMigration
+): Promise<boolean> {
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotificationStart);
+  const answer = await popupMessageNonmodal(versionForMigration);
+  if (answer === learnMoreText) {
+    TOOLS?.ui!.openUrl(learnMoreLink);
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotification, {
+      [TelemetryPropertyKey.status]: TelemetryPropertyValue.learnMore,
+      [TelemetryPropertyKey.mode]: TelemetryPropertyValue.nonmodal,
+    });
+    return false;
+  } else if (answer === upgradeButton) {
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotification, {
+      [TelemetryPropertyKey.status]: TelemetryPropertyValue.ok,
+      [TelemetryPropertyKey.mode]: TelemetryPropertyValue.nonmodal,
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function showConfirmOnlyNotification(ctx: CoreHookContext): Promise<boolean> {
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotificationStart);
+  const res = await TOOLS?.ui.showMessage(
+    "info",
+    getLocalizedString("core.migrationV3.confirmOnly.Message"),
+    true,
+    "OK"
+  );
+  if (res?.isOk() && res.value === "OK") {
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotification, {
+      [TelemetryPropertyKey.status]: TelemetryPropertyValue.ok,
```

Comment:
the property name `status` may not be so accurate, can we use `clickedButton` or something else? Can talk with PM about whether there's a common terminology for this.

Reply:
OK. will use another pr for the telemetry property